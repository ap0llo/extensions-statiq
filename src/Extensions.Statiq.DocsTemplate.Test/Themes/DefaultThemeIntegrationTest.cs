using System.Threading.Tasks;
using AngleSharp.Dom.Html;
using ApprovalTests.Reporters;
using FluentAssertions;
using NUnit.Framework;
using Statiq.Common;
using Statiq.Testing;

namespace Grynwald.Extensions.Statiq.DocsTemplate.Test.Themes
{
    /// <summary>
    /// Rending tests for "Default" theme
    /// </summary>
    [UseReporter(typeof(DiffReporter))]
    public class DefaultThemeIntegrationTest : ThemeIntegrationTestBase
    {
        [Test]
        public async Task Rendering_produces_the_expected_output()
        {
            // ARRANGE
            var testCaseName = $"{nameof(DefaultThemeIntegrationTest)}.{nameof(Rendering_produces_the_expected_output)}";

            var bootstrapper = CreateBootstrapper(DocsTemplateThemeNames.Default, "<p>Some Content</p>")
                .AddSetting(DocsTemplateKeys.SiteName, "Test Site Name");

            // ACT
            var result = await bootstrapper.RunTestAsync();

            // ASSERT
            result.ExitCode.Should().Be(0);
            Approve(testCaseName);
        }

        [TestCase(null, null, "")]
        [TestCase("", "", "")]
        [TestCase(null, "Page Title", "Page Title")]
        [TestCase("", "Page Title", "Page Title")]
        [TestCase("Site Name", null, "Site Name")]
        [TestCase("Site Name", "", "Site Name")]
        [TestCase("Site Name", "Page Title", "Page Title - Site Name")]
        public async Task Output_includes_the_expected_title(string siteName, string pageTitle, string expectedTitle)
        {
            // ARRANGE            
            var bootstrapper = CreateBootstrapper(DocsTemplateThemeNames.Default, "<p>Some Content</p>")
                .AddSetting(DocsTemplateKeys.Title, pageTitle)
                .AddSetting(DocsTemplateKeys.SiteName, siteName);

            // ACT
            var result = await bootstrapper.RunTestAsync();

            // ASSERT
            result.ExitCode.Should().Be(0);

            var output = LoadOutput().ParseAsHtml();

            output.QuerySelectorAll("title")
                .Should().ContainSingle()
                .Which.Should().BeAssignableTo<IHtmlTitleElement>()
                .Which.Text.Should().Be(expectedTitle);
        }
    }
}
