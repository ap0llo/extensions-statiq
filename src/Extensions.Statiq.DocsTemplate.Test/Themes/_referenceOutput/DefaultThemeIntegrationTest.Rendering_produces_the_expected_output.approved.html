<!DOCTYPE html><html lang="en"><head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

<link rel="stylesheet" href="./assets/css/bootstrap/dist/css/bootstrap.min.css">
<link rel="stylesheet" href="./assets/css/highlight.js/styles/default.min.css">
<link rel="stylesheet" href="./assets/css/site.css">


<title>Test Site Name</title>
</head>
<body>

    <header>

<nav class="navbar navbar-expand-lg fixed-top navbar-dark bg-dark shadow">
    <a class="navbar-brand" href="#">Test Site Name</a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse" id="navbarNav">



    </div>
</nav>
    </header>

    <main>
        <div class="container">
            <div class="row">
                <div class="col-md-10">
                    
<p>Some Content</p>
                </div>
            </div>
        </div>
    </main>

    <footer>
    </footer>
<!-- jQuery first, then Popper.js, then Bootstrap JS -->
<script src="./assets/js/jquery/jquery.min.js"></script>
<script src="./assets/js/popper.js/popper.min.js"></script>
<script src="./assets/js/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="./assets/js/highlight.js/highlight.min.js"></script>

<script src="./assets/js/site.js"></script>



</body></html>